src/features/collection/state/collectionState.ts

```diff
@@ -39,25 +43,31 @@ export interface CollectionState {
   // running contains the ids of the currently running workflows in reverse chronological
   // order based on latestRunTime
   running: string[]
+  listedIDs: string[]
+  pendingIDs: string[]
   collectionLoading: boolean
   runningLoading: boolean
+
 }
 
 const initialState: CollectionState = {
   collection: {},
   running: [],
+  listedIDs: [],
+  pendingIDs: [],
   collectionLoading: true,
   runningLoading: true
 }
 
 export const collectionReducer = createReducer(initialState, {
   'API_COLLECTION_REQUEST': (state, action) => {
-    state.collection = {}
+    state.listedIDs = []
     state.collectionLoading = true
   },
   'API_COLLECTION_SUCCESS': (state, action) => {
     action.payload.data.forEach((d: VersionInfo) => {
       state.collection[d.initID] = d
+      state.listedIDs.push(d.initID)
     })
```

Comment:
this leaves open the chance of repeating IDs in the `listedIDs` array. What we need is a set union between current and new values, and the easiest way to do that is to make the `listedIDs` field a `Set<string>` type. You'll need to use `.add` instead of `.push`

```ts
 action.payload.data.forEach((d: VersionInfo) => {
      state.collection[d.initID] = d
      state.listedIDs.add(d.initID)
  })
```